app/assets/javascripts/models/cart.js

```diff
@@ -47,30 +46,41 @@ var removeBook = function(book, userId) {
     data: {
       book_id: book.id,
       user_id: userId
-    }
+    },
+    success: function(response) {
+      toastr.success(book.title + " was removed from your cart!");
+    }.bind(this),
+    error: function (xhr, status, err) {
+      console.error(this.props.url, status, err.toString());
+    }.bind(this)
   }).done(function(message) {
     console.log("Received response for book removal: " + message.message);
   });
-
-  toastr.success(book.title + " removed from your cart!");
 };
 
-var addBook = function(book, userId) {
-  var cartItems = cart.get("items");
-  var books = cartItems.concat([book]);
-  cart.set("items", books);
-
+var addBook = function(book, userId, groups) {
   $.ajax({
     type: "POST",
     url: "/api/v1/carts/add/" + book.id,
     data: {
       book_id: book.id,
-      user_id: userId
+      user_id: userId,
+      groups: groups
+    },
+    success: function(response) {
+      var cartItems = cart.get("items");
+      var books = cartItems.concat([book]);
+      cart.set("items", books);
+      toastr.success(book.title + " was added to your cart!");
+    },
+    error: function(xhr, status, err) {
+      var errors = xhr.responseJSON.errors;
+      for (var error of errors) {
```

Comment:
'for of' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).